dotnet/src/SemanticKernel.Abstractions/Functions/SKJsonSchema.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+#pragma warning disable IDE0130
+// ReSharper disable once CheckNamespace - Using the main namespace
+namespace Microsoft.SemanticKernel;
+
+/// <summary>Represents JSON Schema for describing types used in <see cref="ISKFunction"/>s.</summary>
+[JsonConverter(typeof(SKJsonSchema.JsonConverter))]
+public sealed class SKJsonSchema
+{
+    /// <summary>The schema stored as a string.</summary>
+    private string? _schemaAsString;
+
+    /// <summary>Parses a JSON Schema for a parameter type.</summary>
+    /// <param name="jsonSchema">The JSON Schema as a string.</param>
+    /// <returns>A parsed <see cref="SKJsonSchema"/>.</returns>
+    /// <exception cref="ArgumentException"><paramref name="jsonSchema"/> is null.</exception>
+    /// <exception cref="JsonException">The JSON is invalid.</exception>
+    public static SKJsonSchema Parse(string jsonSchema) => new(JsonSerializer.Deserialize<JsonElement>(jsonSchema));
```

Comment:
Technically, these JsonSerializer methods require reflection to work (even though the particular type uses a built-in converter). At some point this should be updated to use a JsonSerializerContext that includes `JsonElement`.

Reply:
> At some point this should be updated to use a JsonSerializerContext that includes JsonElement.

JsonSerializer is used extensively with reflection throughout SK.  At one point I'd gone through and updated it to use source gen, but it didn't get merged as it was too painful for too little gain at the time.  I'm sure we'll want to revisit it at some point.